app/controllers/requests_controller.rb

```diff
@@ -34,6 +34,12 @@ def cancel
     redirect_back(fallback_location: root_path)
   end
 
+  def delete
+    @request.deleted!
```

Comment:
you should maybe do a `dependent: :destroy` in the has_many association in requests, so that if you delete one request, it deletes automatically its children.